src/ComponentRegister.ts

```diff
@@ -0,0 +1,45 @@
+import { ElementType, ReactNode } from "react";
+
+export interface LinkComponentProps {
+  href: string;
+  className?: string;
+  children: ReactNode;
+}
+
+export type LinkComponent = ElementType<LinkComponentProps>;
+
+export type ComponentRegisterInitOptions = {
+  Link?: LinkComponent;
+};
+
+/**
+ * Allows the user to register components to use with library components.
+ */
+export class ComponentRegister {
+  /**
+   * @static A {@link LinkComponent} to allow `lbh-frontend-react` to hook into routers
+   * or other custom linking libraries. Defaults to `"a"`
+   */
+  static Link: LinkComponent = "a";
+
+  /**
+   * @static Resets the component register
+   */
+  static reset(): void {
+    this.init();
+  }
+
+  /**
+   * @static Defines register attributes as components given by the host
+   * @example
+   * // Instantiates ComponentRegister.Link as TestLinkComponent
+   * ComponentRegister.init({Link: TestLinkComponent});
```

Comment:
```suggestion
   * // To set `lbh-frontend-react` up to use React Router's `Link` component.
   * ComponentRegister.init({
   *   Link: ({ className, href, children }) => (
   *     <ReactRouter.Link className={className} to={href}>{children}</ReactRouter.Link>
   *   )
   * });
```

Let's write examples well formatted with a focus on readability for a user who knows nothing about the internals of this library.